### PR TITLE
Add 1-component `({s,u}{int,norm}{8,16}`, `float16`) and `unorm8x4-bgra` vertex formats

### LINKED ISSUE #6614: 1-component (`{s,u}{int,norm}{8,16}`, `float16`) and `unorm8x4-bgra` vertex formats are missing
**Connections**

- Added in the spec. upstream by [`gpuweb`#4951](https://github.com/gpuweb/gpuweb/pull/4951).
- Firefox downstream tracking: [bug 1933317](https://bugzilla.mozilla.org/show_bug.cgi?id=1933317)

### PR DESCRIPTION
**Pull request by nolanderc**
**Connections**
Closes #6614 

**Description**

When adding the translations from `wgpu::VertexFormat` to the various backends, I did my best to choose the appropriate formats, but I'm not too familiar with DX12 or Metal, so please, take some extra time to verify those.

**Testing**
N/A

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `taplo format`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [x] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [x] Add change to `CHANGELOG.md`. See simple instructions inside file.

### CODE REVIEW

**Review by cwfitzgerald — APPROVED** [review]
As I have a big change coming in that conflicts with this, I'm going to rebase so each commit can land separately.

**Review by cwfitzgerald — DISMISSED** [review]
Ah wait, hold on, metal is a lot more complicated unfortunately, as we don't use normal vertex formats, we manually transform to using vertex pulling.

First step is that I would add tests for all these new formats in https://github.com/gfx-rs/wgpu/blob/e23146aa3e8f5de0820cec108a9c1aa967454a3b/tests/tests/vertex_formats/mod.rs. This can probably join one of the other sets of tests.

Second step is to add support for these formats in naga - https://github.com/gfx-rs/wgpu/blob/314e196c906e779429cae87333010bfc75f6a379/naga/src/back/msl/writer.rs#L4002. If you need mac testing help or any support, please let us know on the wgpu dev matrix in the readme.

I would appreciate the webgpu update as a separate PR, as this has a bit of work left still.

**Comment by nolanderc** [comment]
This should now be good to go @cwfitzgerald @ErichDonGubler

File: naga/src/back/msl/mod.rs
Comment (ErichDonGubler) [inline naga/src/back/msl/mod.rs:320]: question: @gfx-rs/naga: This code has a 5-slot gap for texture indices here. Is there a reason that this gap should continue to exist, given these changes?

File: naga/src/back/msl/writer.rs
Comment (ErichDonGubler) [inline naga/src/back/msl/writer.rs:4010]: question: @gfx-rs/naga: Why do we not re-use the output of `wgpu_core::VertexFormat::size` here? 🤔 I feel like it should be the same in all cases, but I admittedly haven't analyzed for divergences.

**Review by ErichDonGubler — APPROVED** [review]
LGTM. I have questions that I've noted above, but they're not blocking. I will try testing this in CTS on Firefox first before merging, since I have a patch stack ready-but-WIP on the Firefox side. If somebody else needs this, I'm happy to merge earlier.

@cwfitzgerald, did you still want to review this?

File: naga/src/back/msl/mod.rs
Comment (nolanderc) [reply naga/src/back/msl/mod.rs:320]: I noticed the gap in the previous code corresponded to the missing Float64 formats. I suppose having the enum discriminant values be a 1:1 mapping between those in wpgu_types and msl could be yield more efficient conversions between the two, or similar, so I just left it as is.

**Comment by cwfitzgerald** [comment]
I'll take a cursory look.

**Comment by ErichDonGubler** [comment]
I don't think it makes sense to leave this to get stale after being reviewed, so I'm going to merge it. @cwfitzgerald, if you find something you'd like addressed, I'm happy to review a follow-up PR!

**Comment by cwfitzgerald** [comment]
Nah this fell off my radar, merging was the right choice.

**Pull request merged**
